src/azure-cli/azure/cli/command_modules/cdn/_params.py

```diff
@@ -135,6 +135,33 @@ def load_arguments(self, _):
     with self.argument_context('cdn custom-domain create') as c:
         c.argument('location', validator=get_default_location_from_resource_group)
 
+    with self.argument_context('cdn custom-domain enable-https') as c:
+        c.argument('profile_name', id_part=None, help='Name of the parent profile.')
+        c.argument('endpoint_name', help='Name of the parent endpoint.')
+        c.argument('custom_domain_name', name_arg_type, help='Name of the custom domain.')
+        c.argument('min_tls_version',
+                   help='The minimum TLS version required for the custom domain.',
+                   arg_type=get_enum_type(['none', '1.0', '1.2']))
+        c.argument('user_cert_protocol_type',
+                   arg_group='Bring Your Own Certificate',
+                   help='The protocol type of the certificate.',
+                   arg_type=get_enum_type(['sni', 'ip']))
+        c.argument('user_cert_subscription_id',
```

Comment:
is it possible to support one secret id argument?

Reply:
This is a non-trivial request, as we would need to add extra logic to retrieve the correct resource group name for the secret, which is not included in the secret id because it doesn't use a standard Azure resource id. We would also have to correctly handle permissions issues where the user may not have permission to perform that lookup even though they do have permission to reference the key through CDN. Can we include this later as a separate PR if customers request it?